config/routes.rb

```diff
@@ -26,6 +26,7 @@
   get 'complete_post_need' => 'post_needs#complete', :as => :complete_post_need
   get 'undo_complete_post_need' => 'post_needs#undo_complete', :as => :undo_complete_post_need
 
+  get 'need/by_cat', to: 'needs#by_cat', as: :by_cat
```

Comment:
If this controller method gets moved to the model layer, we shouldn't need a route.